hack/test-cmd.sh

```diff
@@ -39,13 +40,22 @@ function start-proxy()
   stop-proxy
 
   kube::log::status "Starting kubectl proxy"
+
+  set-random-port
+
   # the --www and --www-prefix are just to make something definitely show up for
   # wait_for_url to see.
   kubectl proxy -p ${PROXY_PORT} --www=. --www-prefix=/healthz "$@" 1>&2 &
+
   PROXY_PID=$!
   kube::util::wait_for_url "http://127.0.0.1:${PROXY_PORT}/healthz" "kubectl proxy $@"
 }
 
+function set-random-port()
```

Comment:
actually, I changed my mind on this one.

can you instead add as a function in hack/lib/util.sh? I'd like to start using a random port for the etcd we spawn for tests as well, and not rewriting this awk line would be nice.

probably said function should return the port instead of setting an environment variable, though.